src/util/strencodings.cpp

```diff
@@ -526,3 +527,46 @@ std::string HexStr(const Span<const uint8_t> s)
     assert(it == rv.end());
     return rv;
 }
+
+std::optional<uint64_t> ParseByteUnits(const std::string& str, uint64_t default_multiplier)
+{
+    if (str.empty()) {
+        return std::nullopt;
+    }
+    assert(default_multiplier > 0);
+    auto multiplier = default_multiplier;
+    char unit = str.back();
+    switch (unit) {
+    case 'k':
+        multiplier = 1000;
+        break;
+    case 'K':
+        multiplier = 1024;
+        break;
+    case 'm':
+        multiplier = 1000 * 1000;
+        break;
+    case 'M':
+        multiplier = 1ULL << 20;
+        break;
+    case 'g':
+        multiplier = 1000ULL * 1000 * 1000;
+        break;
+    case 'G':
+        multiplier = 1ULL << 30;
+        break;
+    case 't':
+        multiplier = 1000ULL * 1000 * 1000 * 1000;
+        break;
+    case 'T':
+        multiplier = 1ULL << 40;
+        break;
+    default:
+        unit = 0;
+        break;
+    }
+
+    auto parsed_num = ToIntegral<uint64_t>(unit ? str.substr(0, str.size() - 1) : str);
+    // check overflow
+    return !parsed_num || parsed_num > std::numeric_limits<uint64_t>::max() / multiplier ? std::nullopt : std::optional{*parsed_num * multiplier};
```

Comment:
```suggestion
    if( !parsed_num || parsed_num > std::numeric_limits<uint64_t>::max() / multiplier){
    return std::nullopt;
    }
    return *parsed_num * multiplier;
```

Maybe split up into two lines to avoid the `std::optional{}` constructor for the value?

Reply:
I agree, it looks better split